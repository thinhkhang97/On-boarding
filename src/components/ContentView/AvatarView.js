import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Avatar from 'sharedComponents/Avatar';

type Props = {
  source: string
};
const DEFAULT_SIZE = 254;
class AvatarView extends React.Component<Props> {
  render() {
    const { source } = this.props;
    return (
      <View
        style={{
          width: DEFAULT_SIZE,
          height: DEFAULT_SIZE,
          borderRadius: DEFAULT_SIZE / 2,
          borderColor: '#f5f8fc',
          borderWidth: 54,
          overflow: 'hidden',
          padding: 15,
          backgroundColor: 'white',
          marginBottom: 48
        }}
      >
        <Avatar source={source} style={{ flex: 1, width: '100%' }} />
      </View>
    );
  }
}

const styles = StyleSheet.create({});

export default AvatarView;
