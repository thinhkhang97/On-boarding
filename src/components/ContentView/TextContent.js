// @flow
import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Title from '../Title/Title';
import Description from '../Description/Description';
import MixTitle from '../Title/MixTitle';

type Props = {
  mix?: boolean,
  title?: string,
  description?: string,
  mixData?: {
    leftTitle: string,
    leftSubTitle: string,
    rightTitle: string,
    rightSubTitle: string,
  },
  mixTitleStyle?: View.propTypes.style,
  mixSubTitleStyle?: View.propTypes.style,
  titleStyle?: View.propTypes.style,
};

class TextContent extends React.Component<Props> {
  static defaultProps = {
    mix: false,
    mixData: {
      leftTitle: '',
      leftSubTitle: '',
      rightTitle: '',
      rightSubTitle: '',
    },
    title: '',
    description: '',
  };
  render() {
    const {
      mix,
      title,
      description,
      mixData,
      mixTitleStyle,
      mixSubTitleStyle,
      titleStyle,
    } = this.props;
    return (
      <View style={styles.container}>
        {mix ? (
          <View style={styles.titleContainer}>
            <MixTitle
              title={mixData.leftTitle}
              subTitle={mixData.leftSubTitle}
              titleStyle={mixTitleStyle}
              subTitleStyle={mixSubTitleStyle}
            />
            <MixTitle
              title={mixData.rightTitle}
              subTitle={mixData.rightSubTitle}
              titleStyle={mixTitleStyle}
              subTitleStyle={mixSubTitleStyle}
            />
          </View>
        ) : (
          <Title style={[titleStyle, { textAlign: 'center' }]}>{title}</Title>
        )}

        <Description style={styles.descriptionContainer}>{description}</Description>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: { flex: 1, justifyContent: 'space-around' },
  titleContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  descriptionContainer: {
    textAlign: 'center',
    paddingLeft: 39,
    paddingRight: 36,
  },
});

export default TextContent;
