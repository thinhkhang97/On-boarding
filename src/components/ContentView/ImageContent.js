// @flow
// Still have bug with android devices
import * as React from 'react';
import { View, StyleSheet, Image, Dimensions } from 'react-native';
import AvatarView from './AvatarView';

type Props = {
  source: {
    uri: string,
  },
  style?: View.propTypes.style,
  avatar?: boolean,
  imageStyle?: View.propTypes.style,
  avatarBorderColor?: string,
};

const screenWidth = Dimensions.get('window').width;

class ImageContent extends React.Component<Props> {
  static defaultProps = {
    avatar: false,
    style: {},
    imageStyle: {},
  };

  render() {
    const { source, avatar, style, imageStyle, avatarBorderColor } = this.props;
    return (
      <View
        style={[
          !avatar && {
            width: '100%',
            height: '100%',
            paddingTop: 30,
          },
          style,
        ]}
      >
        {avatar ? (
          <AvatarView borderColor={avatarBorderColor} source={source} />
        ) : (
          <Image
            style={[
              {
                width: 'auto',
                height: '100%',
              },
              imageStyle,
            ]}
            source={source}
            resizeMode="contain"
          />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({});

export default ImageContent;
