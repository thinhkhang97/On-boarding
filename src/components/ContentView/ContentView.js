import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import ImageContent from './ImageContent';
import Avatar from 'sharedComponents/Avatar';
import TextContent from './TextContent';

type Props = {
  imageContainer: object
};
const URI =
  'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR2XnXChz496P_XvBFy4e7xp59-y7seeneFBFIYDFnVUQz8U9Fi';
class ContentView extends React.Component<Props> {
  render() {
    const { imageContent, textContent } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <View style={styles.imageContainer}>
          <ImageContent source={{ uri: URI }} />
        </View>
        <View style={styles.textContainer}>
          <TextContent
            mix
            mixData={{
              leftTitle: 3,
              leftSubTitle: 'Projects',
              rightTitle: 90,
              rightSubTitle: 'Connections'
            }}
            description="Nam dapibus nisl vitae elit fringilla rutrum. Aenean sollicitudin, erat a elementum rutrum, neque sem."
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  imageContainer: {
    height: '75%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingBottom: 50
  },
  textContainer: {
    flex: 1
  }
});

export default ContentView;
