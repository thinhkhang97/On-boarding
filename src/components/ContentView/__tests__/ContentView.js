import * as React from 'react';
import renderer from 'react-test-renderer';
import ContentView from '../ContentView';

test('Render avatar view correctly', ()=>{
    const testContentView = renderer.create(<ContentView/>).toJSON();
    expect(testContentView).toMatchSnapshot();
})