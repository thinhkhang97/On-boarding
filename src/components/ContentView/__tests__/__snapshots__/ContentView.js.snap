// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avatar view correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "75%",
        "justifyContent": "flex-end",
      }
    }
  >
    <View />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View />
  </View>
</View>
`;
